ui/src/components/entity/citation/PaperAbstract.tsx

```diff
@@ -0,0 +1,52 @@
+import * as React from "react";
+import { Paper } from "../../../api/types";
+import { truncateText } from "../../../utils/ui";
+
+interface Props {
+  paper: Paper;
+}
+
+interface State {
+  showFullAbstract: boolean;
+}
+
+export default class PaperAbstract extends React.Component<Props, State> {
+  state = {
+    showFullAbstract: false,
+  };
+
+  toggleFullAbstract = () => {
+    this.setState((prev) => {
+      return {
+        showFullAbstract: !prev.showFullAbstract,
+      };
+    });
```

Comment:
```
this.setState(prev => { showFullAbstract: !prev.showFullAbstract });
```

You don't need the `return` if you don't want it.

Reply:
Apparently, I do. If I apply your change, the component stops working.

Reply:
OH! Because I forgot to wrap with `()`.
```
this.setState(prev => ({ showFullAbstract: !prev.showFullAbstract }));
```

![tenor-123380695](https://user-images.githubusercontent.com/39743357/115626348-3a3eba80-a2b2-11eb-9ae6-523d77025b43.gif)